team/bundles/org.eclipse.jsch.ui/src/org/eclipse/jsch/internal/ui/preference/PreferencePage.java

```diff
@@ -497,7 +497,11 @@ else if(e.widget==keyGenerateRSA){
 					final JSchException[] _e=new JSchException[1];
 					BusyIndicator.showWhile(getShell().getDisplay(), () -> {
 						try {
-							_kpair[0] = KeyPair.genKeyPair(getJSch(), __type);
+							if (__type == KeyPair.RSA) {
+								_kpair[0] = KeyPair.genKeyPair(getJSch(), __type, 4096);
+							} else {
+								_kpair[0] = KeyPair.genKeyPair(getJSch(), __type, 3072);
+							}
```

Comment:
As suggested in https://github.com/eclipse-platform/eclipse.platform/issues/1464#issuecomment-2439600327, I think the key-size for RSA and DSA should be specified in a static final int constant at the top of this class:
```
private static final String SSH2_PREFERENCE_PAGE_CONTEXT="org.eclipse.jsch.ui.ssh2_preference_page_context"; //$NON-NLS-1$
private static final int RSA_KEY_SIZE = 4096;
private static final int DSA_KEY_SIZE = 3072;
```

Then you can create a variable in this method, before `BusyIndicator.showWhile()` that holds the actual keySize used for the specified key type:
```
int keySize = type == KeyPair.RSA ? RSA_KEY_SIZE : DSA_KEY_SIZE;
final JSchException[] _e=new JSchException[1];
BusyIndicator.showWhile(getShell().getDisplay(), () -> {
```

and then this if-else-block can stay a one-liner:
```suggestion
							_kpair[0] = KeyPair.genKeyPair(getJSch(), __type, keySize);
```

as well as the block below to create the comment.